controllers/vaultdynamicsecret_controller.go

```diff
@@ -272,6 +268,14 @@ func (r *VaultDynamicSecretReconciler) renewLease(ctx context.Context, c vault.C
 	if err != nil {
 		return nil, err
 	}
+	// The renewal duration can come back as less than the requested increment
+	// if the time remaining on max_ttl is less than the increment. If there are
+	// rollout restart targets and the renewal was capped in this manner, return
+	// an error to ensure the targets have valid credentials while going through
+	// the rollout restart.
+	if len(o.Spec.RolloutRestartTargets) > 0 && resp.LeaseDuration < o.Status.SecretLease.LeaseDuration {
+		return nil, fmt.Errorf("lease renewal was capped to %ds, less than the requested %ds", resp.LeaseDuration, o.Status.SecretLease.LeaseDuration)
```

Comment:
I wonder whether we should do this for all dynamic secrets, not just ones that have rollout restart targets defined? For example if db creds are read from disk (mounted via k8s secret), that pod would equally want a decent grace period where it can load some new credentials before the old ones expire.

Also one uber nit: This will cause a warning event to be emitted, but normal seems more fitting.

Reply:
Yep, all good points. In 523b636 it now handles a truncated lease renewal for any dynamic secret, and emits a more friendly event. Not sure if we want to requeue though, since it's just falling through to refresh the credentials.